flink-core/src/main/java/org/apache/flink/api/connector/source/util/ratelimit/RateLimiterStrategy.java

```diff
@@ -42,9 +43,14 @@ public interface RateLimiterStrategy extends Serializable {
      * @param recordsPerSecond The number of records produced per second. The actual number of
      *     produced records is subject to rounding due to dividing the number of produced records
      *     among the parallel instances.
+     * @deprecated Use {@link PerSecondRateLimiterStrategy#create(double, Duration)}, instead, to
+     *     avoid thread leakage.
      */
+    @Deprecated
     static RateLimiterStrategy perSecond(double recordsPerSecond) {
-        return parallelism -> new GuavaRateLimiter(recordsPerSecond / parallelism);
```

Comment:
Should we not deprecate GuavaRateLimiter.java rather then delete it ?

Reply:
It was explicitly labeled as `@Internal` which is why I went for the removal rather than keeping code duplicates.